pkgs/development/libraries/qt-6/patches/qtdeclarative-invalidate-caches-from-mismatched-store-paths.patch

```diff
@@ -0,0 +1,112 @@
+diff --git a/src/qml/jsruntime/qv4executablecompilationunit.cpp b/src/qml/jsruntime/qv4executablecompilationunit.cpp
+index 2ab38b70d9..8634b07e41 100644
+--- a/src/qml/jsruntime/qv4executablecompilationunit.cpp
++++ b/src/qml/jsruntime/qv4executablecompilationunit.cpp
+@@ -33,6 +33,8 @@
+ #include <QtCore/qcryptographichash.h>
+ #include <QtCore/QScopedValueRollback>
+ 
++#include <QtCore/qcoreapplication.h>
++
+ static_assert(QV4::CompiledData::QmlCompileHashSpace > QML_COMPILE_HASH_LENGTH);
+ 
+ #if defined(QML_COMPILE_HASH) && defined(QML_COMPILE_HASH_LENGTH) && QML_COMPILE_HASH_LENGTH > 0
+@@ -44,6 +46,36 @@ __attribute__((section(".qml_compile_hash")))
+ const char qml_compile_hash[QV4::CompiledData::QmlCompileHashSpace] = QML_COMPILE_HASH;
+ static_assert(sizeof(QV4::CompiledData::Unit::libraryVersionHash) > QML_COMPILE_HASH_LENGTH,
+               "Compile hash length exceeds reserved size in data structure. Please adjust and bump the format version");
++
++bool nix__isNixApplication() {
++  static const bool value = QCoreApplication::applicationFilePath().startsWith(QStringLiteral("/nix/store/"));
```

Comment:
Nit: this isn't entirely correct because the store path can be overridden, but meh, it's hardcoded in a bunch more places anyway.

Reply:
Also: oof magic statics, I hate C++.

Reply:
`replaceVars` + `builtins.storeDir` would be the thing to do here.